encoder.go

```diff
@@ -11,31 +11,113 @@ import (
 // Signed
 
 func EncodeInt8(w io.Writer, i int8) (err error) {
-	return EncodeVarint(w, int64(i))
+	return EncodeVarint(w, uint64(i))
 }
 
 func EncodeInt16(w io.Writer, i int16) (err error) {
-	return EncodeVarint(w, int64(i))
+	return EncodeVarint(w, uint64(i))
 }
 
 func EncodeInt32(w io.Writer, i int32) (err error) {
 	var buf [4]byte
-	binary.BigEndian.PutUint32(buf[:], uint32(i))
+	binary.LittleEndian.PutUint32(buf[:], uint32(i))
 	_, err = w.Write(buf[:])
 	return
 }
 
 func EncodeInt64(w io.Writer, i int64) (err error) {
 	var buf [8]byte
-	binary.BigEndian.PutUint64(buf[:], uint64(i))
+	binary.LittleEndian.PutUint64(buf[:], uint64(i))
 	_, err = w.Write(buf[:])
 	return err
 }
 
-func EncodeVarint(w io.Writer, i int64) (err error) {
-	var buf [10]byte
-	n := binary.PutVarint(buf[:], i)
-	_, err = w.Write(buf[0:n])
+func EncodeVarint(w io.Writer, v uint64) (err error) {
```

Comment:
This is the encoding function for Proto3 unsigned varints.  Signed varints should use the previous logic, binary.PutVarint()...